src/app/services/app-auth.service.ts

```diff
@@ -0,0 +1,71 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export type User = {
+  id: string;
+  name: string;
+  balance: number;
+};
+
+/**
+ * This is not the actual implementation of authentication service,
+ * this is only example, be careful with reading all of the notes inside.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AppAuthService {
+  private isAuthenticated$$ = new BehaviorSubject<boolean>(false);
+
+  private currentUser$$ = new BehaviorSubject<User | null>(null);
+  public currentUser$ = this.currentUser$$.asObservable();
+
+  /**
+   * In the production version of auth service it should be like
+   * public isAuthenticated$ = this.authServiceFromFoundation.isAuthenticated;
+   */
+  public isAuthenticated$ = this.isAuthenticated$$.asObservable();
+
+  /**
+   * In the production version of app auth service there will be only one dependency
+   * for authService from foundation-ang
+   */
+  constructor(private http: HttpClient, private router: Router) {}
+
+  /**
+   * In the production version only `redirectUri` is needed to be provided as an argument for
+   * this.authServiceFromFoundation.login({ redirectUri }) method call,
+   * and the authentication request will be handled by identity provided from the settings
+   */
+  login({ login, redirectUri }: { login: string; redirectUri: string }): void {
+    this.http
```

Comment:
you should add an error handler for 401 response. Also I don't think you should just redirect from here. But also return something to confirm that login was successful or not.